test/unit/ip-utils.spec.ts

```diff
@@ -0,0 +1,558 @@
+      })
+    })
+
+    describe('and both IPs are "unknown"', () => {
+      beforeEach(() => {
+        originalIp = 'unknown'
+        currentIp = 'unknown'
+      })
+
+      it('should allow access', () => {
+        const result = validateIpAddress(originalIp, currentIp)
+
+        expect(result.valid).toBe(true)
+      })
+    })
+  })
+
+  describe('and validation should fail', () => {
+    describe('and current IP is "unknown" but original is not', () => {
+      beforeEach(() => {
+        originalIp = '203.0.113.1'
+        currentIp = 'unknown'
+      })
+
+      it('should return valid as false', () => {
+        const result = validateIpAddress(originalIp, currentIp)
+
+        expect(result.valid).toBe(false)
+      })
+
+      it('should provide unable to verify IP message', () => {
+        const result = validateIpAddress(originalIp, currentIp)
+
+        expect(result.reason).toBe('Unable to verify IP address')
+      })
+
+      it('should provide current_ip_unknown metric reason', () => {
+        const result = validateIpAddress(originalIp, currentIp)
+
+        expect(result.metricReason).toBe('current_ip_unknown')
+      })
+    })
+
+    describe('and IPs do not match', () => {
+      beforeEach(() => {
+        originalIp = '203.0.113.1'
+        currentIp = '203.0.113.2'
+      })
+
+      it('should return valid as false', () => {
+        const result = validateIpAddress(originalIp, currentIp)
+
+        expect(result.valid).toBe(false)
+      })
+
+      it('should provide detailed mismatch message', () => {
+        const result = validateIpAddress(originalIp, currentIp)
+
+        expect(result.reason).toBe('IP address mismatch. Original: 203.0.113.1, Current: 203.0.113.2')
+      })
+
+      it('should provide ip_mismatch metric reason', () => {
+        const result = validateIpAddress(originalIp, currentIp)
+
+        expect(result.metricReason).toBe('ip_mismatch')
+      })
+    })
+  })
+
+  describe('and edge cases are handled', () => {
+    describe('and IPv6 addresses are used', () => {
+      let ipv6Address: string
+
+      beforeEach(() => {
+        ipv6Address = '2001:0db8:85a3:0000:0000:8a2e:0370:7334'
+        originalIp = ipv6Address
+        currentIp = ipv6Address
+      })
+
+      it('should handle IPv6 addresses correctly', () => {
+        const result = validateIpAddress(originalIp, currentIp)
+
+        expect(result.valid).toBe(true)
+      })
+    })
+
+    describe('and IPv6 vs IPv4 mismatch occurs', () => {
+      beforeEach(() => {
+        originalIp = '203.0.113.1'
+        currentIp = '2001:0db8:85a3::8a2e:0370:7334'
+      })
+
+      it('should detect mismatch between IPv4 and IPv6 addresses', () => {
+        const result = validateIpAddress(originalIp, currentIp)
+
+        expect(result.valid).toBe(false)
+        expect(result.metricReason).toBe('ip_mismatch')
+      })
+    })
+
+    describe('and IP comparison is case sensitive', () => {
```

Comment:
As IPs are always numbers or symbols there shouldn't be any issues with issues with cases sensitiveness, what do you think about removing this tests?